src/main/java/org/assertj/core/api/recursive/comparison/RecursiveComparisonConfiguration.java

```diff
@@ -57,6 +57,12 @@
   private Set<FieldLocation> ignoredCollectionOrderInFields = new LinkedHashSet<>();
   private List<Pattern> ignoredCollectionOrderInFieldsMatchingRegexes = new ArrayList<>();
 
+  // RecursiveIterableComparisonAssert configuration section
+  private boolean ignoreActualIterableOrder = false; // ignore the order of iterables when comparing their elements
+  private boolean strictTypeCheckingOnActualIterable = false; // don't allow different type of iterables to be compared (i.e. List vs Set)
+  private boolean allowArrayTypeForExpected = false;
```

Comment:
We need to describe this properties in `multiLineDescription` but that will impact `RecursiveComparisonAssert` assertion. 

Thinking out loud, I believe we only need `RecursiveComparisonConfiguration` for both `RecursiveComparisonAssert` and `RecursiveIterableComparisonAssert` as `RecursiveComparisonAssert` must be able to compare iterable fields.

Reply:
It means the field must not named only for `RecursiveIterableComparisonAssert`, hopefully me renaming suggestions in the method exposing these fields applies to both recursive assert classes.

Reply:
How about I remove `ignoreActualIterableOrder` and `strictTypeCheckingOnActualIterable` and use `ignoreCollectionOrder` and `strictTypeChecking` respectively for both purposes as you suggested earlier ?

Reply:
That the idea, as I said in a previous comment what applies for field should also work for top level objects.